terraform/launchpad-token-dispenser-lambda.tf

```diff
@@ -43,4 +43,137 @@ resource "aws_ssm_parameter" "launchpad_token_dispenser_lambda_arn" {
+resource "aws_iam_policy" "tds_cloudtrail_to_cloudwatch_policy" {
+  name        = "${var.prefix}-cloudtrail-to-cloudwatch-policy"
+  description = "Policy for CloudTrail to write logs to CloudWatch Logs"
+
+  policy = jsonencode({
+    Version = "2012-10-17",
+    Statement = [
+      {
+        Effect = "Allow",
+        Action = [
+          "logs:CreateLogStream",
+          "logs:PutLogEvents"
+        ],
+        Resource = [
+          "${aws_cloudwatch_log_group.tds_cloudtrail_log_group.arn}:*"
+        ]
+      }
+    ]
+  })
+}
+
+# Attach the policy to the role
+resource "aws_iam_role_policy_attachment" "tds_cloudtrail_to_cloudwatch_attachment" {
+  role       = aws_iam_role.tds_cloudtrail_to_cloudwatch_role.name
+  policy_arn = aws_iam_policy.tds_cloudtrail_to_cloudwatch_policy.arn
+}
+
+resource "aws_cloudtrail" "launchpad_token_dispenser_trail" {
+  depends_on = [ aws_cloudwatch_log_group.tds_cloudtrail_log_group ]
+  name                          = "${var.prefix}-launchpad-token-dispenser-trail"
+  s3_bucket_name                = aws_s3_bucket.cloudtrail_bucket.id
+  include_global_service_events = true
+  is_multi_region_trail         = true
+  enable_logging                = true
+
+  # Bind cloudtrail to the CloudWatch Logs group
+  cloud_watch_logs_group_arn = "${aws_cloudwatch_log_group.tds_cloudtrail_log_group.arn}:*"
+  cloud_watch_logs_role_arn  = aws_iam_role.tds_cloudtrail_to_cloudwatch_role.arn
+
+  event_selector {
+    read_write_type           = "All"
+    include_management_events = false
+
+    data_resource {
+      type   = "AWS::Lambda::Function"
+      values = [aws_lambda_function.launchpad_token_dispenser_lambda.arn]
+    } 
+  }
+}
+
+
+# S3 bucket for CloudTrail logs
+resource "aws_s3_bucket" "cloudtrail_bucket" {
+  bucket = "${var.prefix}-cloudtrail-logs"
+
+  acl = "private"
+
+  versioning {
+    enabled = false
+  }
+
+  lifecycle {
+    prevent_destroy = true
+  }
+
+  # Add lifecycle rule to expire objects after 7 days
+  lifecycle_rule {
+    id      = "delete-cloudtrail-logs"
+    enabled = true
+
+    expiration {
+      days = 7
+    }
+  }
+}
+
+# S3 bucket policy for CloudTrail
+resource "aws_s3_bucket_policy" "tds_cloudtrail_bucket_policy" {
+  bucket = aws_s3_bucket.cloudtrail_bucket.id
+
+  policy = jsonencode({
+    Version = "2012-10-17",
+    Statement = [
+      {
+        Sid: "AWSCloudTrailWrite20150309",
+        Effect = "Allow",
+        Principal = {
+          Service = "cloudtrail.amazonaws.com"
+        },
+        Action = "s3:GetBucketAcl",
+        Resource = "arn:aws:s3:::${aws_s3_bucket.cloudtrail_bucket.id}"
+      },
+      {
+        Sid: "AWSCloudTrailWrite",
+        Effect = "Allow",
+        Principal = {
+          Service = "cloudtrail.amazonaws.com"
+        },
+        Action = "s3:PutObject",
+        Resource = "arn:aws:s3:::${aws_s3_bucket.cloudtrail_bucket.id}/AWSLogs/${data.aws_caller_identity.current.account_id}/*",
```

Comment:
Can we change `arn:aws:s3:::${aws_s3_bucket.cloudtrail_bucket.id}` to `aws_s3_bucket.cloudtrail_bucket.arn`?